File: 860-lemonade-change/860-lemonade-change.java
class Solution {
    public boolean lemonadeChange(int[] bills) {
        int tens = 0, fives = 0;
        for(int bill : bills){
            if(bill == 5){
                fives += 1;
            } else if(bill == 10){
                tens += 1;
                fives -= 1;
                if(fives < 0){
                    return false;
                }
            } else {
                if(tens > 0){
                    tens -= 1;
                    fives -= 1;
                } else {
                    fives -= 3;
                }
                
                if(tens < 0 || fives < 0){
                    return false;
                }
            }
        }
        return true;
    }
}

File: 860-lemonade-change/860-lemonade-change.py
class Solution:
    def lemonadeChange(self, bills: List[int]) -> bool:
        fives = tens = 0
        for bill in bills:
            if bill == 5:
                fives += 1
            elif bill == 10:
                tens += 1
                fives -= 1
                if fives < 0:
                    return False
            else:
                if tens > 0:
                    tens -= 1
                    fives -= 1
                else:
                    fives -= 3
                    
                if tens < 0 or fives < 0:
                    return False
        return True